addons/project/models/project.py

```diff
@@ -1050,7 +1058,19 @@ def default_get(self, default_fields):
     @api.model_create_multi
     def create(self, vals_list):
         default_stage = dict()
+        default_parent_stage = dict()
         for vals in vals_list:
+            parent_id = vals.get('parent_id')
+            if parent_id and "stage_id" not in vals:
+                # Sub-task stage_id must by default be the same as its parent stage_id
+                # 1) It must override project default stage
+                # 2) Ensures the default is only computed once per parent task
+                if parent_id not in default_parent_stage:
+                    parent_task = self.env['project.task'].browse(parent_id)
```

Comment:
calling ``browse`` in a ``for`` loop is not very efficient. 

You could map the data before and call browse only once. 

Example (not tested though).

``` python3
parent_ids = [vals['parent_id'] for vals in vals_list if vals.get('parent_id')]
mapped_stages = {t: t.stage_id for t in t in self.env['project.task'].browse(parent_ids)}
```